charts/gitops-server/templates/deployment.yaml

```diff
@@ -65,6 +84,30 @@ spec:
           {{- end }}
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
+          {{- if or .Values.serverTLS.useSecret .Values.serverTLS.useLocalFiles }}
+          volumeMounts:
+            - name: tls-volume
+              readOnly: true
+              mountPath: "/etc/tls-volume"
+          {{- end }}
+      {{- if .Values.serverTLS.useSecret }}
+      volumes:
+        - name: tls-volume
+          secret:
+            secretName: {{ .Values.serverTLS.secretName }}
```

Comment:
can we add some docs for how that secret should look? and also include an example template?

Reply:
k8s has an `tls` secret type which we should use https://kubernetes.io/docs/concepts/configuration/secret/#tls-secrets

Reply:
working on this now...

Reply:
@SamLR @Callisto13  I wonder if we should replace the NOT TLS secret type approach (option 1) with the TLS secret type instead? What do you guys think will be the most likely scenario? Will it be worth supporting both approaches?

Reply:
I think having one approach is sufficient, the TLS secret type feels like the winner to me